packages/ts/react-form/src/index.ts

```diff
@@ -85,56 +101,96 @@ function getBinderNodeControls<T, M extends AbstractModel<T>>(
   };
 }
 
-function useFields(): FieldDirective {
-  const registry = useRef(new WeakMap<HTMLElement, AbstractFieldStrategy>());
-
-  return (model) => {
-    const n = getBinderNode(model);
-
-    let fld: HTMLElement | null;
+function useFields<T, M extends AbstractModel<T>>(node: BinderNode<T, M>): FieldDirective {
+  return useMemo(() => {
+    const registry = new WeakMap<AbstractModel<any>, FieldState<any>>();
+
+    return ((model: AbstractModel<any>) => {
+      const n = getBinderNode(model);
+
+      const fieldState: FieldState<unknown> = registry.has(model)
+        ? registry.get(model)!
+        : {
+          value: undefined,
+          required: false,
+          invalid: false,
+          errorMessage: '',
+          strategy: undefined,
+        };
+
+      if (!registry.has(model)) {
+        registry.set(model, fieldState);
+      }
 
-    const updateValueEvent = () => {
-      if (fld) {
-        if (!isFieldElement(fld)) {
-          throw new TypeError(`Element '${fld.localName}' is not a form element`);
+      if (fieldState.strategy) {
+        const valueFromField = convertFieldValue(model, fieldState.value);
+        if (valueFromField !== n.value && !(Number.isNaN(n.value) && Number.isNaN(valueFromField))) {
+          fieldState.value = n.value;
+          fieldState.strategy.value = n.value;
         }
 
-        let strategy = registry.current.get(fld);
-
-        if (!strategy || strategy.model !== model) {
-          strategy = getDefaultFieldStrategy(fld, model);
-          registry.current.set(fld, strategy);
+        if (fieldState.required !== n.required) {
+          fieldState.required = n.required;
+          fieldState.strategy.required = n.required;
         }
 
-        const elementValue = strategy.value;
+        const firstError: ValueError<any> | undefined = n.ownErrors[0];
```

Comment:
```suggestion
        const [firstError] = n.ownErrors;
```

Reply:
or we could also use:
```suggestion
        const firstError = n.ownErrors.at(0);
```